src/rauc-installer.c

```diff
@@ -134,14 +134,20 @@ static struct install_context *install_context_new(void)
  */
 static void install_context_free(struct install_context *context)
 {
-        if (context == NULL)
+        GMainContext *loop_context = NULL;
+
+        if (!context)
                 return;
+
+        loop_context = g_main_loop_get_context(context->mainloop);
+
         g_free(context->bundle);
         g_mutex_clear(&context->status_mutex);
         g_main_context_unref(context->loop_context);
         g_assert_cmpint(context->status_result, >=, 0);
         g_assert_true(g_queue_is_empty(&context->status_messages));
         g_main_loop_unref(context->mainloop);
+        g_main_context_unref(loop_context);
```

Comment:
isn't this the same context we freed in line 146 with `g_main_context_unref(context->loop_context);` already? Or did I miss something?

The creation says:

``` c
GMainContext *loop_context = g_main_context_new();
[...]
context->mainloop = g_main_loop_new(loop_context, FALSE);
context->loop_context = loop_context;
```

Reply:
You're right, I somehow missed that. Will fix it.